experimental/offline_attestation/server/src/main.rs

```diff
@@ -0,0 +1,134 @@
+use clap::Parser;
+use log::{debug, info};
+use offline_attestation_shared::{
+    decrypt, encrypt, generate_private_key, init, EncryptedRequest, EncryptedResponse, Handle,
+    PublicKeyInfo,
+};
+use std::{
+    net::{Ipv6Addr, SocketAddr},
+    sync::Arc,
+};
+use warp::{hyper::StatusCode, reply::Response, Filter, Reply};
+
+const ECHO_PATH: &str = "encrypted_echo";
+
+const PUBLIC_KEY_INFO_PATH: &str = "public_key_info";
+
+#[derive(Parser, Clone, Debug)]
+#[clap(about = "Offline Attestation Server")]
+pub struct Opt {
+    #[clap(long, default_value = "8080", help = "Port number on which to listen.")]
+    port: u16,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    env_logger::init();
+    init();
+    let opt = Opt::parse();
+    let address = SocketAddr::from((Ipv6Addr::UNSPECIFIED, opt.port));
+
+    // Generate a private key for the lifetime of the server at startup.
+    // In a production implementation there would typically be some form of key provisioning
+    // mechanism that will ensure all servers that run the right code in an appropriate enclave
+    // receive the same key.
+    debug!("Generating private key");
+    let private_key_handle = Arc::new(generate_private_key()?);
+    let public_key_handle = private_key_handle
+        .public()
+        .map_err(|error| anyhow!("Couldn't get public key: {}", error))?;
+
+    // Combine the public key with an attesation report in preparation for publishing it.
+    let public_key_info = Arc::new(PublicKeyInfo::new(&public_key_handle)?);
+    debug!(
+        "Constructed public key info: {}",
+        serde_json::to_string(public_key_info.as_ref())?
+    );
+
+    // Set up the path filters and appropriate handlers.
+    let root_filer = create_root_filter(private_key_handle, public_key_info);
+
+    // Start serving requests.
+    warp::serve(root_filer).run(address).await;
+    Ok(())
+}
+
+/// Creates the root filter that combines the two path filters into a single one.
+fn create_root_filter(
+    private_key_handle: Arc<Handle>,
+    public_key_info: Arc<PublicKeyInfo>,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    get_public_key_info(public_key_info).or(encrypted_echo(private_key_handle))
+}
+
+/// Path filter for "GET /public_key_info".
+///
+/// Returns the JSON-encoded public key info.
+fn get_public_key_info(
+    public_key_info: Arc<PublicKeyInfo>,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    warp::get()
+        .and(warp::path(PUBLIC_KEY_INFO_PATH))
+        .and(warp::path::end())
+        .map(move || warp::reply::json(public_key_info.clone().as_ref()))
+}
+
+/// Path filter for "POST /encrypted_echo" with a JSON body.
+fn encrypted_echo(
+    private_key_handle: Arc<Handle>,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+    warp::post()
+        .and(warp::path(ECHO_PATH))
+        // Limit the maximum request body size to 4kb.
```

Comment:
```suggestion
        // Limit the maximum request body size to 4KiB.
```

(I think)

namely:

- Ki = 1024 vs k = 1000
- B = bytes vs b = bits